### ISSUE TITLE
bring back IP and/or device name

### ISSUE THREAD

**Issue by DJCrashdummy** [cowbell/sharedrop#71]
IIRC some time ago the internal IP and device name (if available) was used to distinguish the members of the network.

now the animal-avatars make it funny and easy for tech-unsavvy users, but if you are familiar with the network and have reasonable device names, it got unnecessary hard to identify the correct participant (and tedious because they change at every reload of the page).

may i suggest to add the IP and/or device name (at least one durable identifier) in brackets to the avatars again!?
--> so it will be both: easy & funny for newbies and straight forward for techies with fixed IPs or reasonable names in the network.

**Comment by szimek** [comment]
The issue is that browsers stopped exposing local IP addresses, because of security reasons, so it became really unreliable. More info here; https://tools.ietf.org/html/draft-mdns-ice-candidates-00.

Adding a device name might make sense, e.g. https://snapdrop.net/ does it. Regarding keeping identifier persistent between page loads - it should be possible when using uuid, because it's very unlikely that 2 different devices will receive exactly the same one, but because currently there's no logic to unsure that 2 users won't end up with the same animal avatar, I can't really make these persistent.

Let me know if persistent uuid would be enough.

**Comment by DJCrashdummy** [comment]
> The issue is that browsers stopped exposing local IP addresses, because of security reasons, so it became really unreliable.

yes, i always block it via uBlockO and unblock it when needed for conference calls, file sharing etc.

> Adding a device name might make sense, e.g. https://snapdrop.net/ does it.

well, "device names" on its own aren't a good solution either... although they are persistent, they will quickly cause confusion in a network with more than a few devices (especially at homogeneous environments like offices, classroom etc. - see RobinLinus/snapdrop/issues/13 & RobinLinus/snapdrop/issues/69).

> keeping identifier persistent between page loads - it should be possible when using uuid [...]

it isn't also the perfect solution, but i guess it can make things less tedious... because once "identified" a device, it stays the same.

---

IMHO if using IP addresses is impossible, how about hostnames? - i know they are also not 100% reliable, so i guess a combination of measures would be the best solution... an attempt to collect & prioritize them:

1. keep the animal avatars (by default)
  --> to keep the process a no-brainer for tech-unsavvy users
0. add hostnames in brackets, like it was once in addition to the IPs
  --> for straight forward identification if reasonable hostnames are available
0. make (animal) avatars persistent
  --> to make identification efficient, if sharedrop is used more than once (in a while)
0. make the animal avatars resp. its text to be pecise editable (at least on/for the own machine)
  --> to easily note and don't have to remember the identified machines

**Comment by DuBistKomisch** [comment]
Would be nice to at least be able to see unreliable IP address as well as the new names
